packages/game-passes/test/GamePasses.test.ts

```diff
@@ -0,0 +1,1251 @@
+      expect(await sandboxPasses.supportsInterface(AccessControlInterfaceId)).to
+        .be.true;
+    });
+  });
+
+  describe('Event Emissions', function () {
+    it('should emit TokenConfigured event when configuring a token', async function () {
+      const NEW_TOKEN_ID = 5;
+
+      await expect(
+        sandboxPasses
+          .connect(admin)
+          .configureToken(
+            NEW_TOKEN_ID,
+            true,
+            200,
+            20,
+            'ipfs://QmNewToken',
+            user1.address,
+          ),
+      )
+        .to.emit(sandboxPasses, 'TokenConfigured')
+        .withArgs(
+          NEW_TOKEN_ID,
+          true,
+          200,
+          20,
+          'ipfs://QmNewToken',
+          user1.address,
+        );
+    });
+
+    it('should emit TokenConfigUpdated event when updating a token', async function () {
+      await expect(
+        sandboxPasses.connect(admin).updateTokenConfig(
+          TOKEN_ID_1,
+          200, // new max supply
+          15, // new max per wallet
+          'ipfs://QmUpdated',
+          user2.address,
+        ),
+      )
+        .to.emit(sandboxPasses, 'TokenConfigUpdated')
+        .withArgs(TOKEN_ID_1, 200, 15, 'ipfs://QmUpdated', user2.address);
+    });
+
+    it('should emit TransferabilityUpdated event when changing transferability', async function () {
+      await expect(
+        sandboxPasses.connect(admin).setTransferable(TOKEN_ID_1, false),
+      )
+        .to.emit(sandboxPasses, 'TransferabilityUpdated')
+        .withArgs(TOKEN_ID_1, false);
+    });
+
+    it('should emit TransferWhitelistUpdated event when updating whitelist', async function () {
+      await expect(
+        sandboxPasses
+          .connect(admin)
+          .updateTransferWhitelist(TOKEN_ID_1, [user1.address], true),
+      )
+        .to.emit(sandboxPasses, 'TransferWhitelistUpdated')
+        .withArgs(TOKEN_ID_1, user1.address, true);
+    });
+
+    it('should emit BaseURISet event when updating base URI', async function () {
+      const newBaseURI = 'https://new-api.example.com/metadata/';
+
+      await expect(sandboxPasses.connect(admin).setBaseURI(newBaseURI))
+        .to.emit(sandboxPasses, 'BaseURISet')
+        .withArgs(BASE_URI, newBaseURI);
+    });
+
+    it('should emit TokensRecovered event when recovering tokens', async function () {
+      const mockToken = await deployToken();
+      const amount = ethers.parseEther('10');
+      await mockToken.mint(await sandboxPasses.getAddress(), amount);
+
+      await expect(
+        sandboxPasses
+          .connect(admin)
+          .recoverERC20(await mockToken.getAddress(), treasury.address, amount),
+      )
+        .to.emit(sandboxPasses, 'TokensRecovered')
+        .withArgs(await mockToken.getAddress(), treasury.address, amount);
+    });
+  });
+
+  describe('Token Recovery', function () {
+    it('should allow admin to recover ERC20 tokens', async function () {
```

Comment:
missing cannot recover tokens if not admin

Reply:
Resolved in an earlier comment, all admin functions are now test for non-admin.